tox.ini

```diff
@@ -3,9 +3,9 @@ skipsdist = True
 usedevelop = True
 
 envlist =
-    python{3.7,3.8,3.9,3.10}-django{3.2}-wagtail{2.15,2.16,3.0,4.0,main}-{sqlite,postgres}
-    python{3.8,3.9,3.10}-django{4.0}-wagtail{2.16,3.0,4.0,main}-{sqlite,postgres}
-    python{3.8,3.9,3.10}-django{4.1,main}-wagtail{4.0,main}-{sqlite,postgres}
+    python{3.7}-django{3.2}-wagtail{4.1,4.2,main}-{sqlite,postgres}
+    python{3.8,3.9,3.10}-django{4.0,4.1}-wagtail{4.1,4.2,main}-{sqlite,postgres}
+    python{3.11}-django{4.1,main}-wagtail{4.1,4.2,main}-{sqlite,postgres}
```

Comment:
Suggestion:
- Create two postgres environments
- Test different combinations of python, django and wagtail

```
    python{3.7}-django{3.2}-wagtail{4.1,4.2,main}-{sqlite,postgres12}
    python{3.8,3.9,3.10,3.11}-django{3.2,4.0,4.1,main}-wagtail{4.1,4.2,main}-sqlite
    python{3.8,3.9,3.10}-wagtail{4.1,4.2,main}-{postgres12,postgres15}
    python{3.8,3.9,3.10}-wagtail{3.2}-postgres12
    python{3.11}-wagtail{4.1,4.2,main}-postgres15

deps =
    ...

    postgres12: psycopg2>=2.6,<2.9.5
    postgres15: psycopg2>=2.9.5
```
As suggested from this conversation: https://torchbox.slack.com/archives/C03R0FNK7/p1675655322697539
Reference: https://github.com/wagtail/wagtail-localize/pull/673/files#r1097252879

Reply:
Apparently, newer versions of postgres (15+) work with Python 3.11 but not with older Django (3.2); older versions of postgres (10 - 12) do not work with Python 3.11 but work with Django 3.2.
So tests with Django 3.2 are isolated.